status/assign_roles.py

```diff
@@ -0,0 +1,89 @@
+import tornado.web
+import json
+import time
+import copy
+import base64
+import requests
+
+from dateutil import parser
+from status.util import SafeHandler, UnsafeHandler
+from status.authorization import UnAuthorizedHandler
+
+
+class AssignRolesHandler(SafeHandler):
+    """ Serves a page with users and roles listed, with the option to create new users
+    URL: /assign_roles
+    """
+    def get(self):
+        t = self.application.loader.load("assign_roles.html")
+        self.write(t.generate(gs_globals=self.application.gs_globals, user=self.get_current_user(),
+        roles=self.application.genstat_defaults['roles']))
+
+
+class AssignRolesUsersHandler(SafeHandler):
+    """Serves the data for populating user roles table and also methods to modify user roles
+    URL: /api/v1/assign_roles/users
+    """
+    def get(self):
+        self.set_header("Content-type", "application/json")
+        view_result={}
+        for row in self.application.gs_users_db.view('authorized/roles'):
+            view_result[row.key]=row.value
+        self.write(view_result)
+
+    def post(self):
+        data = json.loads(self.request.body)
+        userToChange = data['username']
+        headers = {"Accept": "application/json",
```

Comment:
Is this really necessary? I would think this would be taken care of by the couchdb python wrapper?

Reply:
Not sure, so left it in.